protocol/value_test.go

```diff
@@ -76,3 +81,166 @@ func vmap(kt, vt wire.Type, items ...wire.MapItem) wire.Value {
 func vitem(k, v wire.Value) wire.MapItem {
 	return wire.MapItem{Key: k, Value: v}
 }
+
+// testValue wraps around wire.Value to allow for stream encoding/decoding
+// testing.
+type testValue struct {
```

Comment:
This seems like binary.Writer implemented in terms of binary.StreamWriter. We should just do that in the actual implementation and then we don't necessarily need this.